core/src/main/java/jenkins/model/DownloadSettings.java

```diff
@@ -116,8 +117,17 @@ public DailyCheck() {
                 }
             }
             if (!due) {
+                // JENKINS-32886: downloadables like the tool installer data may have never been tried if the plugin
+                // was installed "after a restart", so let's give them a try here.
+                final long now = System.currentTimeMillis();
```

Comment:
Just in case, better to wrap the code by <code>catch (Exception ex)</code>. Downloadables can come from the external plugin, and a NPE there may impact the startup flow